src/testhost.x86/DefaultEngineInvoker.cs

```diff
@@ -62,14 +60,19 @@ public void Invoke(IDictionary<string, string> argsDictionary)
                 // Attach to exit of parent process
                 var parentProcessId = CommandLineArgumentsHelper.GetIntArgFromDict(argsDictionary, ParentProcessIdArgument);
                 EqtTrace.Info("DefaultEngineInvoker: Monitoring parent process with id: '{0}'", parentProcessId);
-                var processHelper = new ProcessHelper();
-                processHelper.SetExitCallback(
-                    parentProcessId,
-                    () =>
-                        {
-                            EqtTrace.Info("DefaultEngineInvoker: ParentProcess '{0}' Exited.", parentProcessId);
-                            Environment.Exit(1);
-                        });
+
+                // In remote scenario we cannot monitor parent process, so we expect user to pass parentProcessId as -1
```

Comment:
In that case, is it the runtime providers responsibility to clean up the stale test host processes? Should it be called out explicitly somewhere?